python/rmm/rmm.py

```diff
@@ -268,7 +252,11 @@ def rmm_cupy_allocator(nbytes):
     if cupy is None:
         raise ModuleNotFoundError("No module named 'cupy'")
 
-    return cupy.cuda.memory.MemoryPointer(RMMCuPyMemory(nbytes), 0)
+    buf = librmm.device_buffer.DeviceBuffer(size=nbytes)
+    mem = cupy.cuda.UnownedMemory(ptr=buf.ptr, size=buf.size, owner=buf)
```

Comment:
To support empty arrays, we should set the device id.
```python
dev_id = -1 if buf.ptr else cupy.cuda.device.get_device_id()
mem = cupy.cuda.UnownedMemory(ptr=buf.ptr, size=buf.size, owner=buf, device_id=dev_id)
```
But let's only set it on empty arrays so we don't impose wrong IDs on non-null pointers

Reply:
Thanks Mads! Should be fixed in the latest commit 🙂